src/js/components/agency/visualizations/objectClass/MajorObjectClasses.jsx

```diff
@@ -48,9 +48,9 @@ export default class MajorObjectClasses extends React.Component {
         window.addEventListener('resize', this.handleWindowResize);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.majorObjectClasses.children.length > 0) {
-            this.buildTree(nextProps);
+    componentDidUpdate(prevProps) {
+        if (prevProps.majorObjectClasses.children.length > 0) {
+            this.buildTree(prevProps);
```

Comment:
we should never be using `prevProps` except to confirm the props have changed. So in this case, buildTree should be given `this.props`